packages/gatekeeper/src/components/login.tsx

```diff
@@ -1,34 +1,71 @@
 import React from 'react';
+import { OauthLoginProps } from '..';
+import { NO_PROVIDERS_STRING, OAUTH_LOGIN_PROMPT_STRING } from '../helpers/constants';
 import { getProviderFromOptions, Providers } from '../helpers/oauth';
 
-/** interface for OauthLogin props */
-export interface OauthLoginProps {
-  ProviderLinksComponent?: React.ElementType;
+/** types of supported authorization grant flow */
+export enum AuthorizationGrantType {
+  IMPLICIT = 'IMPLICIT',
+  AUTHORIZATION_CODE = 'AUTHORIZATION_CODE'
+}
+
+/** describes object to pass to the OAuthHook */
+export interface OAuthLoginHookOptions {
   providers: Providers;
+  authorizationGrant: AuthorizationGrantType;
 }
 
 /** interface for ProviderLinks props */
-export interface ProviderLinksProps {
-  providers: Providers;
+export interface ProviderLinksProps extends OAuthLoginHookOptions {
+  OAUTH_LOGIN_PROMPT?: string;
+}
+
+/** interface for OauthLogin props */
+export interface OauthLoginProps extends ProviderLinksProps {
+  ProviderLinksComponent?: React.ElementType;
+  NO_PROVIDERS?: string;
+}
```

Comment:
I think you should not mark these properties as optional.  I think of it this way: 

- can this property make any sense with the optional properties excluded
- is it justifiable to "weaken" the type definition like this?
- for React components, Typescript works with "defaultProps" to negate the need to mark anything as optional e.g. here https://github.com/onaio/js-tools/blob/master/packages/gatekeeper/src/components/callback.tsx#L29

Reply:
Made them required.